src/main/java/ladder/domain/ladder/LadderGenerator.java

```diff
@@ -0,0 +1,19 @@
+package ladder.domain.ladder;
+
+import ladder.domain.generator.BooleanGenerator;
+import ladder.domain.player.Players;
+
+public class LadderGenerator {
+    private final BooleanGenerator booleanGenerator;
+
+    public LadderGenerator(final BooleanGenerator booleanGenerator) {
+        this.booleanGenerator = booleanGenerator;
+    }
+
+    public Ladder generate(final Players players, final Height height) {
+        final int playerCount = players.countPlayers();
```

Comment:
뽀로로가 해준 것처럼 players를 받는 방법과
player 갯수를 받는 방법이 있었을 것 같은데
전자를 선택한 이유가 있을까요?

Reply:
메서드에서 변수명을 굳이 반복해서 쓸 필요는 없어보여요, count 내지는 다른 용어로 바꾸는 것은 어떤가요?
```suggestion
    public Ladder generate(final Players players, final Height height) {
        final int playerCount = players.count();
```

Reply:
그렇네요!! 👍👍 `countPlayers` -> `count` 로 메서드명 수정하였습니다.

---

>뽀로로가 해준 것처럼 players를 받는 방법과
player 갯수를 받는 방법이 있었을 것 같은데
전자를 선택한 이유가 있을까요?

제가 전자를 선택한 이유에 대해 2가지를 설명할 수 있을 거 같아요!! 🧐

1. 가독성
```java
public class LadderController {
    ....

    public void run() {
        final Players players = retryOnException(this::readPlayers);
        final Height height = retryOnException(this::readLadderHeight);

        final int playerCount = players.count();
        final int ladderHeight = height.getHeight();
        final Ladder ladder = ladderGenerator.generate(playerCount, ladderHeight);

        printLadder(players, ladder);
    }
```
위와 같이 아서가 말씀해준 다른 방법에 대한 코드를 작성해보았어요! 😀

이 방식도 충분히 좋다 느껴지지만, 실제적으로 `playerCount`와 `ladderHeight`를 사용하는 곳이 중복되지도 않고, 한번만 사용하므로 `controller`에서는 단순히 포장된 `Players`와 `Height`를 객체 그대로 전달하는 것이 가독성 측면에서 좋다 느껴졌어요!! 😀

2. LadderGenerator 클래스 역할
```java
public class LadderGenerator {
    ...

    public Ladder generate(final int playerCount, final int ladderHeight) {
        return new Ladder(playerCount, ladderHeight, booleanGenerator);
    }
```
`LadderGenerate`의 `generate` 메서드에서 단순히 플레이어의 수와 사다리 높이를 전달받으면, `generate` 메서드는 다음과 같이 바로 `Ladder` 클래스를 새롭게 생성하고 종료되게 됩니다.🥲
그러면, `LadderGenerator`의 역할이 더욱 크게 의미가 없어지는 것 같아 전자를 선택하였습니다!! 😀

아서가 생각했을 때, 어떤 방법이 더 좋은 거 같나요?? 🧐
아니면 더 나은 방법에 대한 아이디어가 있나요?? 아서의 생각이 궁금해요~~!! 🧐🧐🧐

Reply:
뽀로로의 생각을 들려주셔서 감사합니다, 충분히 일리있는 주장이라고 생각해요. 👍🏻
그런데 사고 확장을 위해서 반박하는 입장을 드려볼텐데 잘 고민해보시고, 
이해가 안되면 언제든지 DM 바랍니다!

1. 가독성 측면

> 실제적으로 playerCount와 ladderHeight를 사용하는 곳이 중복되지도 않고, 한번만 사용하므로 controller에서는 단순히 포장된 Players와 Height를 객체 그대로 전달하는 것이 가독성 측면에서 좋다 느껴졌어요!! 

우선 중복으로 사용하지 않고 한번만 사용하는 부분은 변수 선언을 하지 않고 값을 바로 넘겨줌으로 해결할 수 있을 것 같아요. 그럼 포장하는 비용이 들지 않습니다. 

`Players`를 전달했을 때는 외부에서는 `Ladder`가 생성될 때 `Players` 라는 객체에서 뭔가를 하는구나 하고 유추할 수 밖에 없습니다. 그런데 인원수를 count해서 넘겨주면 `아~ 사람의 인원수만 알면 되는구나` 한눈에 파악가능하죠. 그럼 가독성 측면에서도 더 좋은 거 아닐까요?

2. LadderGenerator 클래스의 역할

> 그러면, LadderGenerator의 역할이 더욱 크게 의미가 없어지는 것 같아 전자를 선택하였습니다!! 😀

그럼 더 좋은거 아닌가요? 애초에 필요없었던 객체였는데 의미 부여를 하기 위해서 더 어려운 방식을 선택할 필요는 없을 것 같은데요.
Ladder의 생성자를 바로 이용해줘서 생성할 때 거치는 객체를 줄여도 되서 더 좋을 것 같은데요?

---

혹시 몰라서 한번 더 말씀드립니다.
적용 안하셔도 됩니다! 저는 뽀로로가 작성한 방식도 마음에 들어요ㅎㅎ
단지 이런 입장이 있을수도 있는데 어떻게 지켜나갈 것인지?
본인의 생각을 고수할 것인지? 받아들일 것인지 한번 고민해보시길 바랍니다!

Reply:
너무 좋은 피드백인거 같아요ㅠㅠㅠ 👍👍👍
확실히 아서의 입장을 들어보니 사고 확장이 된다는 느낌이 들었어요!!! 😀

2단계를 진행하면서 끊임없이 고민하고, 더 나은 방향으로 리팩토링 해보겠습니다!! 
너무 감사해요!! 😀